aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,126 @@
+| `SupportedProtocols` | Protocols supported by this hub. By default, all protocols registered on the server are allowed, but protocols can be removed from this list to disable specific protocols for individual hubs. |
+| `EnableDetailedErrors` | If true, sends detailed error messages to the client when exceptions occur. The detailed error messages may contain sensitive data, and are `false` by default. |
+
+Options can be configured for all hubs by providing an options delegate to the `AddSignalR` call in `ConfigureServices`.
+
+[!code-csharp[Startup](configuration/sample/config-startup.cs?range=7-14)]
+
+Options for a single hub override the global options provided in `AddSignalR`, and can be configured using `AddHubOptions<T>`:
+
+[!code-csharp[HubOptions](configuration/sample/config-startup.cs?range=16-19)]
+
+### HttpConnectionDispatcherOptions - Options related to the transport layer.
+
+Options related to the transport layer. Use these to restrict the transports that can be used by SignalR clients, as well as to configure advanced settings related to memory buffer management. The transport options are configured by passing a delegate to `MapHub<T>`.
+
+| Option | Description |
+| ------ | ----------- |
+| `ApplicationMaxBufferSize`  | The maximum number of bytes the connection (transport) can buffer. It buffers when invoking methods from the client, before blocking and waiting for the application to allow writing again. |
+| `AuthorizationData` | A pre-populated list of `IAuthorizeData` gathered from the `Authorize` attributes used on the `Hub<T>`. |
+| `LongPolling`  | Gets the LongPolling options object that has a settable `PollTimeout` property. Setting the `PollTimeOut` sets the wait time before ending a poll request. |
+| `TransportMaxBufferSize`  | The maximum number of bytes the application. For example, `Clients.All.SendAsync` can buffer when writing to a single connection before blocking and waiting for the connection to consume enough bytes to allow writing again. |
+| `Transports`  | A bitmask that sets a transport. The available `options.Transports` options are as follows: `HttpTransportType.WebSockets`, `HttpTransportType.LongPolling`, and `HttpTransportType.ServerSentEvents` |
+| `WebSockets`*  |  Gets the WebSockets options object.  |
+| `CloseTimeout`*  | TimeSpan to set how long to wait for a clean WebSocket close when connection is being terminated.  |
+| `SubProtocolSelector`* | A delegate the hub calls and passes a list of `Sec-WebSocket-Protocol` values from the request header. A delegate returns the chosen `Sec-WebSocket-Protocol`. |
+| `AccessTokenProvider`* | A delegate that is called to get an access token. The token is applied as an HTTP Bearer Authentication header. |
+
+ Items in the table marked with an asterisk (*) are specific to WebSockets.
+
+### HubConnectionBuilder
```

Comment:
"Configuring Client Options"

Reply:
This section looks really rough right now. I think we should split it into two, with one for .NET and one for JavaScript. There will be some duplication, but I think that's OK.

In each section, I think we should list the properties on the relevant object (JS: https://github.com/aspnet/SignalR/blob/dev/clients/ts/signalr/src/IHttpConnectionOptions.ts, C#: https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnectionOptions.cs) with the descriptions we've got here. Code samples can come afterwards, illustrating some of the scenarios.